### PR TITLE
Add intial translation of the figures for the chapters of graph, searching, and sorting

### PR DESCRIPTION
**Pull request by krahets**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [x] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [x] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [ ] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [ ] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [ ] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

**Comment by krahets** [comment]
Please review the translation based on the following aspects:

Ensure the expression is authentic and accurate.
Maintain consistency between the Chinese and English versions. (Balancing with point 1)
In addition to content, please pay attention to the format:

It is expected to use Roboto font for normal text and Roboto Mono for formulas and code. These may be prone to mistakes since they are set manually.
The style and color of shapes and texts should be consistent between the two versions. (I made some color adjustments in the EN version to enhance readability, so please only report those that feel uncomfortable to you.)

File: en/docs/chapter_graph/graph_operations.assets/adjacency_list_step2_add_edge.png
Comment (K3v123) [inline en/docs/chapter_graph/graph_operations.assets/adjacency_list_step2_add_edge.png:1]: text are not bold in en ver. same for below

File: en/docs/chapter_sorting/counting_sort.assets/counting_sort_step7.png
Comment (K3v123) [inline en/docs/chapter_sorting/counting_sort.assets/counting_sort_step7.png:1]: eng ver not bold

File: en/docs/chapter_sorting/merge_sort.assets/merge_sort_step1.png
Comment (K3v123) [inline en/docs/chapter_sorting/merge_sort.assets/merge_sort_step1.png:1]: not bold, same as below images

File: en/docs/chapter_sorting/selection_sort.assets/selection_sort_step1.png
Comment (K3v123) [inline en/docs/chapter_sorting/selection_sort.assets/selection_sort_step1.png:1]: the "1st" is bold in CN same as the "5th" in below and other images etc

**Review by K3v123 — COMMENTED** [review]
overall looks good to me.
just some bolds text again in EN ver

**Review by RafaelCaso — APPROVED** [review]
Looks good (no typos)
My only input is in bubble_sort_overview.png and selection_sort_step11.png we say,
"remaining one element"

I understand we're saying, the element that has a value of 1, correct? It might be a little confusing. Maybe use quotation marks around one (remaining "one" element) but I think that looks a little silly. Maybe format it so it sticks out more? I'd be curious what other reviewers think of this.

File: en/docs/chapter_sorting/bubble_sort.assets/bubble_sort_overview.png
Comment (RafaelCaso) [inline en/docs/chapter_sorting/bubble_sort.assets/bubble_sort_overview.png:1]: "remaining one element..." seems a little confusing. Curious what others think about this

File: en/docs/chapter_sorting/selection_sort.assets/selection_sort_step11.png
Comment (RafaelCaso) [inline en/docs/chapter_sorting/selection_sort.assets/selection_sort_step11.png:1]: "remaining one element..." seems a little confusing. Curious what others think about this

File: en/docs/chapter_sorting/insertion_sort.assets/insertion_operation.png
Comment (pengchzn) [inline en/docs/chapter_sorting/insertion_sort.assets/insertion_operation.png:1]: Temporarily store **base**

File: en/docs/chapter_sorting/heap_sort.assets/heap_sort_step1.png
Comment (pengchzn) [inline en/docs/chapter_sorting/heap_sort.assets/heap_sort_step1.png:1]: The content of these two pics seem different

File: en/docs/chapter_sorting/heap_sort.assets/heap_sort_step10.png
Comment (pengchzn) [inline en/docs/chapter_sorting/heap_sort.assets/heap_sort_step10.png:1]: Lack the translation of "Valid interval"

File: en/docs/chapter_sorting/heap_sort.assets/heap_sort_step1.png
Comment (pengchzn) [reply en/docs/chapter_sorting/heap_sort.assets/heap_sort_step1.png:1]: “Insert data into the heap” may be better, the same goes for the following

File: en/docs/chapter_sorting/heap_sort.assets/heap_sort_step12.png
Comment (pengchzn) [inline en/docs/chapter_sorting/heap_sort.assets/heap_sort_step12.png:1]: The shaded area is incorrect.

File: en/docs/chapter_sorting/bubble_sort.assets/bubble_sort_overview.png
Comment (pengchzn) [reply en/docs/chapter_sorting/bubble_sort.assets/bubble_sort_overview.png:1]: (remaining one element does not need sorting) -> (no need to sort the remaining element)

File: en/docs/chapter_searching/replace_linear_by_hashing.assets/two_sum_brute_force.png
Comment (pengchzn) [inline en/docs/chapter_searching/replace_linear_by_hashing.assets/two_sum_brute_force.png:1]: when a combination with the sum equal to target.

File: en/docs/chapter_searching/binary_search_insertion.assets/binary_search_insertion_naive.png
Comment (pengchzn) [inline en/docs/chapter_searching/binary_search_insertion.assets/binary_search_insertion_naive.png:1]: use binary search to find any 6
use linear search to find...

File: en/docs/chapter_sorting/heap_sort.assets/heap_sort_step10.png
Comment (krahets) [reply en/docs/chapter_sorting/heap_sort.assets/heap_sort_step10.png:1]: No space for that

File: en/docs/chapter_sorting/heap_sort.assets/heap_sort_step1.png
Comment (krahets) [reply en/docs/chapter_sorting/heap_sort.assets/heap_sort_step1.png:1]: > The content of these two pics seem different

The Chinese version had been updated.

File: en/docs/chapter_sorting/summary.assets/sorting_algorithms_comparison.png
[the remark this answers is not in the document]
Comment (krahets) [reply en/docs/chapter_sorting/summary.assets/sorting_algorithms_comparison.png:1]: No space for it

**Comment by krahets** [comment]
Thanks for your thorough review! @pengchzn, @RafaelCaso, @K3v123, most of the comments have been addressed. A few comments remain unresolved for further discussion. Please let me know if there's anything else that needs fixing.

**Pull request merged**